fabric-item-group-api-v1/src/main/java/net/fabricmc/fabric/mixin/itemgroup/ItemGroupMixin.java

```diff
@@ -41,13 +45,42 @@ public class ItemGroupMixin implements IdentifiableItemGroup {
 	@Final
 	private int index;
 
-	@Inject(method = "getStacks", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemGroup;addItems(Lnet/minecraft/resource/featuretoggle/FeatureSet;Lnet/minecraft/item/ItemGroup$Entries;)V", shift = At.Shift.AFTER), locals = LocalCapture.CAPTURE_FAILHARD)
-	public void getStacks(FeatureSet featureSet, boolean search, CallbackInfoReturnable<ItemStackSet> cir, ItemGroup.EntriesImpl entries) {
-		final Event<ItemGroupEvents.ModifyEntries> modifyEntriesEvent = ItemGroupEventsImpl.getModifyEntriesEvent(getId());
+	@Shadow
+	private ItemStackSet displayStacks;
+
+	@Shadow
+	private ItemStackSet searchTabStacks;
+
+	@SuppressWarnings("ConstantConditions")
+	@Inject(method = "getStacks", at = @At(value = "FIELD", target = "Lnet/minecraft/item/ItemGroup;searchTabStacks:Lnet/minecraft/item/ItemStackSet;", opcode = Opcodes.PUTFIELD, shift = At.Shift.AFTER))
+	public void getStacks(FeatureSet featureSet, boolean search, CallbackInfoReturnable<ItemStackSet> cir) {
+		// Sanity check for the injection point. It should be after these fields are set.
+		Objects.requireNonNull(displayStacks, "displayStacks");
+		Objects.requireNonNull(searchTabStacks, "searchTabStacks");
+
+		// Convert the entries to lists
+		var mutableDisplayStacks = new LinkedList<>(displayStacks);
+		var mutableSearchTabStacks = new LinkedList<>(searchTabStacks);
+		var content = new ItemGroupContent(featureSet, mutableDisplayStacks, mutableSearchTabStacks);
+
+		final Event<ItemGroupEvents.ModifyContent> modifyEntriesEvent = ItemGroupEventsImpl.getModifyEntriesEvent(getId());
 
 		if (modifyEntriesEvent != null) {
-			modifyEntriesEvent.invoker().modifyItems(featureSet, entries);
+			modifyEntriesEvent.invoker().modifyContent(featureSet, content);
```

Comment:
Why aren't we passing the search field here couldn't that be useful for modders?